### PR TITLE
Add ragged gather reduce v2

### PR DESCRIPTION
**Pull request by NuojCheng**
# Description

Onboard the `ragged_gather_reduce_v2` kernel to MaxText. 

# Performance Evaluation

## Synthetic Data (Random Routing, EP=8)

* **Before:** `xids/271201875` | [xprof](http://xprof/?session_id=chengnuojin-13723506073515044095)
* **After:** `xids/271194893` | [xprof](http://xprof/?session_id=chengnuojin-16980623064568288653)

**Observations:**
* Step time improved, dropping from 14.24s to 13.17s.
* Unsort kernel time per layer dropped significantly from 13.97ms to 4.9ms.
* Sort kernel time remained unchanged.

## Real Data (Without Random Routing, EP=4)

* **Before:** `xids/271240396` | [xprof](http://xprof/?session_id=chengnuojin-7050341447920488795)
* **After:** `xids/271240858` | [xprof](http://xprof/?session_id=chengnuojin-17666397907218336633)

**Observations:**
* Step time improved slightly, dropping from 17.8s to 17.5s.
* Unsort kernel time remained unchanged.

## Update Unit Test

This PR also updates emb dim from 2048 to 7168 to better reflect this feature's performance on DSv3. The testing time for this change is small

### Before

```
49.58s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts
41.99s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts_ragged_buffer
38.16s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts
34.38s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts_ragged_buffer
29.79s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts_fallback
16.93s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts_fallback
```

### After

```
53.06s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts
37.33s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts_ragged_buffer
35.49s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts
28.26s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts_ragged_buffer
25.93s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_no_ring_of_experts_fallback
13.38s call     tests/unit/moe_test.py::RoutedMoeTest::test_ragged_sort_loss_and_grad_ring_of_experts_fallback
```

# Tests

* **Convergence:** Tested in `xids/271420711`. The loss decreases normally.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):

- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by RissyRan — APPROVED** [review]
Thanks!

File: src/maxtext/kernels/ragged/ragged_sort.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import jax
 import jax.numpy as jnp
 from maxtext.kernels.ragged.ragged_gather import ragged_gather
-from maxtext.kernels.ragged.ragged_gather_reduce import ragged_gather_reduce
+from maxtext.kernels.ragged.ragged_gather_reduce_v2 import ragged_gather_reduce
```
Comment (gobbleturk) [inline src/maxtext/kernels/ragged/ragged_sort.py:20]: should we remove the ragged_gather_reduce file?

File: tests/unit/moe_test.py [hunk 2]
```diff
@@ -660,7 +660,7 @@ def _build_cfg(use_ragged_sort: bool):
           enable_checkpointing=False,
           model_name="mixtral-8x7b",
           override_model_config=True,
-          base_emb_dim=2048,  # we want emb dim being multiple of 1024 for fully using the kernel
+          base_emb_dim=7168,  # we want emb dim being multiple of 1024 for fully using the kernel
```
Comment (gobbleturk) [inline tests/unit/moe_test.py:663]: why make this change?

**Review by gobbleturk — APPROVED** [review]

File: src/maxtext/kernels/ragged/ragged_sort.py (hunk 1 above)
Comment (NuojCheng) [reply src/maxtext/kernels/ragged/ragged_sort.py:20]: let's keep it for now in case we find issues with the new kernel. I will deprecate it in future PR (like GMM v1 that we should deprecate)

File: tests/unit/moe_test.py (hunk 2 above)
Comment (NuojCheng) [reply tests/unit/moe_test.py:663]: To make the test closer to DSv3..

Comment (NuojCheng) [reply tests/unit/moe_test.py:663]: And in PR description this change does not increase test time much
